### ISSUE TITLE
Total tokens not updated

### ISSUE THREAD

**Issue by dexturr** [vegaprotocol/vega#6532]
# Problem encountered
After voting on a proposal the total tokens is not update. Perviously this only worked through the single proposal API, now it doesn't appear to work at all.

# Observed behaviour
The votes are present but they are not counted towards the total tokens number

# Expected behaviour
Total tokens should be a sum of all the parties votes in that direction 

# System response

```js
{
  "proposal": {
    "id": "eba76a48c3dcad1c04a564feb189d924897860103c804aca33fc95916bed4e76",
    "rationale": {
      "title": "test",
      "description": "test proposal",
      "__typename": "ProposalRationale"
    },
    "reference": "",
    "state": "STATE_OPEN",
    "datetime": "2022-10-17T15:03:14.422435Z",
    "rejectionReason": null,
    "party": {
      "id": "f9982447fb4128f9968f9981612c5ea85d19b62058ec2636efc812dcbbc745ca",
      "__typename": "Party"
    },
    "errorDetails": null,
    "terms": {
      "closingDatetime": "2022-10-19T16:03:07Z",
      "enactmentDatetime": null,
      "change": {
        "__typename": "NewFreeform"
      },
      "__typename": "ProposalTerms"
    },
    "votes": {
      "yes": {
        "totalTokens": "0", // <<<<<<<< HERE
        "totalNumber": "2",
        "votes": [
          {
            "value": "VALUE_YES",
            "party": {
              "id": "c443fb0388266c290736f325efeaad9aaa6d5f7f7b184f4e2302ecf8207b056e",
              "stakingSummary": {
                "currentStakeAvailable": "4000000000000000000",
                "__typename": "StakingSummary"
              },
              "__typename": "Party"
            },
            "datetime": "2022-10-18T09:24:43.242774Z",
            "__typename": "Vote"
          },
          {
            "value": "VALUE_YES",
            "party": {
              "id": "f9982447fb4128f9968f9981612c5ea85d19b62058ec2636efc812dcbbc745ca",
              "stakingSummary": {
                "currentStakeAvailable": "1000000000000000000",
                "__typename": "StakingSummary"
              },
              "__typename": "Party"
            },
            "datetime": "2022-10-17T15:21:51.78078Z",
            "__typename": "Vote"
          }
        ],
        "__typename": "ProposalVoteSide"
      },
      "no": {
        "totalTokens": "0",
        "totalNumber": "0",
        "votes": null,
        "__typename": "ProposalVoteSide"
      },
      "__typename": "ProposalVotes"
    },
    "__typename": "Proposal"
  }
}
```

# Steps to reproduce

## Manual
Steps to reproduce the behaviour manually:
1. Create a proposal
2. Vote on proposal
3. Notice totalTokens in direction you voted in are still 0

## Automation
Link to automation and explanation on how to run it to reproduce the problem/bug

# Evidence

## Logs
If applicable, add logs and/or screenshots to help explain your problem.

## Additional context
Add any other context about the problem here including; system version numbers, components affected.

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

**Comment by dexturr** [comment]
Possibly related:
https://github.com/vegaprotocol/vega/issues/6356
https://github.com/vegaprotocol/vega/issues/6504

**Comment by EVODelavega** [comment]
We only work out the token totals when the voting is closed. As far as the core is concerned, the totals are meaningless before that point. If we were to keep track of the totals, each vote cast would result in us checking if the party has changed their mind (replacing a previous Y/N vote with the opposite), and updating the totals. Instead, the values are kept blank (nil/zero) until we actually check whether the proposal was accepted or declined.

TL;DR

This is expected behaviour for open proposals
